pkg/autoscaler/autoscaler.go

```diff
@@ -457,6 +459,64 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 				cleanupObjects[asr.GetName()] = asr
 			}
 
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Waiting for MachineSet %s to acquire autoscaling annotations", machineSet.GetName()))
+				Eventually(func() (bool, error) {
+					current, err := framework.GetMachineSet(client, machineSet.GetName())
+					if err != nil {
+						return false, err
+					}
+
+					annotations := current.GetAnnotations()
+					_, minFound := annotations[caMinSizeAnnotation]
+					_, maxFound := annotations[caMaxSizeAnnotation]
+					return minFound && maxFound, nil
+				}, framework.WaitShort, pollingInterval).Should(BeTrue(), "MachineSet %s did not acquire autoscaling annotations", machineSet.GetName())
+			}
+
+			By("Waiting 1 minute to give the new nodes extra time to register their resource capacity")
+			time.Sleep(1 * time.Minute)
+
+			// on some cloud providers, we have experienced the new nodes having resources added to their status.capacity
+			// after they have been initialized. this has been seen with `hugepages-1Gi` and `hugepages-2Mi`. so we
+			// wait until the new nodes have the same resource keys before progressing.
+			By("Waiting for the new Nodes to have similar resources")
+			Eventually(func() (bool, error) {
+				nodes1, err := framework.GetNodesFromMachineSet(client, transientMachineSets[0])
+				if err != nil {
+					return false, err
+				}
+				if len(nodes1) != 1 {
+					return false, fmt.Errorf("Got more nodes than expected for MachineSet %s", transientMachineSets[0].GetName())
```

Comment:
`more` is not necessarily true, you could have got no node right?

Reply:
good point

Reply:
i ended up refactoring this code to be a little less repetitive